helix-common/src/main/java/org/apache/helix/HelixConfigStringProcessUtil.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class HelixConfigStringProcessUtil {
```

Comment:
Nit: when I see this class, the name suggests to me that this is a util class that processes all Helix utils... where in reality, I think you meant for this to be used only for parsing InstanceConfigs, and zooming further into this, not all fields in InstanceConfig, but specifically the string that goes into the "domain" only.

As such, I think that it might improve readability and minimize confusion if you renamed it to something more descriptive - something like `InstanceDomainStringParser` or `InstanceDomainStringUtil`?

Lastly, a block comment above `public final class` that briefly describes what this does and why it's needed? (or more along the lines of what the domain in InstanceConfig is used for).

Reply:
Thanks for the suggestion. 
As our offline discussion, let's keep the class and method name more generic as it will be used for cluster config in following PR. 
Changed the naming according to convention.